### PR TITLE
Use switch instead of if statement in editor_node.cpp

### PR DESCRIPTION
**Pull request by NilsIrl**
Use a switch statement instead of many if statements.

Reasons for:
1. Follows the rest of the code base
2. Prevents `p_what` from being checked at each `if` statement uselessly

### CODE REVIEW

File: editor/editor_node.cpp [hunk 1]
```diff
@@ -238,185 +238,182 @@ void EditorNode::_unhandled_input(const Ref<InputEvent> &p_event) {
...
-		_update_update_spinner();
-	}
+			_update_update_spinner();
+		} break;
 
-	if (p_what == Control::NOTIFICATION_RESIZED) {
-		_update_scene_tabs();
+		case Control::NOTIFICATION_RESIZED: {
+			_update_scene_tabs();
+		}
```
Comment (Xrayez) [inline editor/editor_node.cpp]: Missing break statement?
```suggestion
		} break;
```
More notifications might be added so worth to break it anyway.

File: editor/editor_node.cpp [hunk 2]
```diff
@@ -238,185 +238,182 @@ void EditorNode::_unhandled_input(const Ref<InputEvent> &p_event) {
...
-		editor_data.save_editor_external_data();
-		FileAccess::set_file_close_fail_notify_callback(NULL);
-		log->deinit(); // do not get messages anymore
-	}
-	if (p_what == NOTIFICATION_PROCESS) {
-
-		if (opening_prev && !confirmation->is_visible())
-			opening_prev = false;
+	switch (p_what) {
+		case NOTIFICATION_EXIT_TREE: {
```
Comment (akien-mga) [inline editor/editor_node.cpp]: Could you move `EXIT_TREE` below `ENTER_TREE`? That would be a more natural order.

File: editor/editor_node.cpp [hunk 3]
```diff
@@ -238,185 +238,182 @@ void EditorNode::_unhandled_input(const Ref<InputEvent> &p_event) {
...
 
-		ResourceImporterTexture::get_singleton()->update_imports();
-	}
-	if (p_what == NOTIFICATION_ENTER_TREE) {
+			scene_root->set_size_override(true, Size2(ProjectSettings::get_singleton()->get("display/window/size/width"), ProjectSettings::get_singleton()->get("display/window/size/height")));
 
-		Engine::get_singleton()->set_editor_hint(true);
+			ResourceImporterTexture::get_singleton()->update_imports();
+		} break;
+		case NOTIFICATION_ENTER_TREE: {
```
Comment (akien-mga) [inline editor/editor_node.cpp]: Maybe add an empty line before this one, as done for other cases? That does make the code slightly more readable IMO with such big code blocks.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**